web/src/message_list_view.js

```diff
@@ -527,6 +527,16 @@ export class MessageListView {
             }
         };
 
+        // when we are building message_group for a single message before
+        // adding it to another group, the prev needs to be initialized before
+        // the loop so that message_container's calculated variables are properly
+        // initialized.
+        // This prev can also be undefined in case the message group being
+        // built just has a single message.
+        if (message_containers.length === 1) {
+            prev = this._message_groups.at(-1)?.message_containers.at(-1);
+        }
```

Comment:
Hmm, but is this wrong if we're prepending, which is calculated in `merge_message_groups` / `join_message_groups`. (If we're adding new messages at the top, it seems likely wrong?).

I'm still a bit puzzled; `this.set_calculated_message_container_variables(message_container)` is not a function of `prev`.

Reply:
well, I guess this is involving just "building a message group". It wouldn't actually affect anything because currently, if `message_containers` size is 1, then `prev` is false for sure.

Reply:
Well, I guess you're right....in this case we're just going with the assumption that we're gonna attach this message group at the end of another message group....So in that case, maybe we need to deal with `set_calculated_message_container_variables(message_container)` for group of existing `message_containers`, and `message_container` of live messages we receive separately 🤔

Reply:
I posted a suggestion in Zulip, but I think this particular set of variables just needs to be computed after the build/join/merge process is complete.

Reply:
thanks! I will update it.